git/refs/log.py

```diff
@@ -217,9 +217,9 @@ def entry_at(cls, filepath, index):
             the index is negative
         """
         fp = open(filepath, 'rb')
-        if index < 0:
-            return RefLogEntry.from_line(fp.readlines()[index].strip())
-        else:
+        with open(filepath, 'rb') as fp:
```

Comment:
This looks like it will open the file twice, leaking one file handle in the process. The first `open` occours in the line above.

Reply:
Yeah, Thanks